miri-script/src/commands.rs

```diff
@@ -0,0 +1,711 @@
+        let sysroot = self.extra_vars.sysroot.to_string_lossy().to_owned();
+        install_to_dir(
+            &self.miri_dir.to_string_lossy().to_owned(),
+            &toolchain,
+            &self.cargo_extra_flags,
+            &sysroot,
+            &args,
+        )?;
+        let cargo_miri_dir = path!(self.miri_dir / "cargo-miri");
+        install_to_dir(
+            &cargo_miri_dir.to_string_lossy().to_owned(),
+            &toolchain,
+            &self.cargo_extra_flags,
+            &sysroot,
+            &args,
+        )?;
+        Ok(())
+    }
+
+    fn build_package(
+        // Path to Cargo.toml file of a package to build.
+        path: &str,
+        toolchain: &str,
+        extra_flags: &str,
+        args: &[&str],
+    ) -> Result<()> {
+        cmd(
+            "cargo",
+            chain!(
+                [&toolchain, "build", extra_flags, "--manifest-path", path,],
+                args.into_iter().cloned()
+            ),
+        )
+        .run()?;
+        Ok(())
+    }
+
+    fn build(&self, flags: &[String]) -> Result<()> {
+        // Build, and let caller control flags.
+        let toolchain = self.get_toolchain_override();
+        let miri_manifest = path!(self.miri_dir / "Cargo.toml");
+        let cargo_miri_manifest = path!(self.miri_dir / "cargo-miri" / "Cargo.toml");
+        let args = flags.iter().map(String::as_str).collect::<Vec<_>>();
+        Self::build_package(
+            &miri_manifest.to_string_lossy().to_owned(),
+            &toolchain,
+            &self.cargo_extra_flags,
+            &args,
+        )?;
+        Self::build_package(
+            &cargo_miri_manifest.to_string_lossy().to_owned(),
+            &toolchain,
+            &self.cargo_extra_flags,
+            &args,
+        )?;
+        Ok(())
+    }
+
+    fn check(&self, flags: &[String]) -> Result<()> {
+        fn check_package(
+            // Path to Cargo.toml file of a package to check.
+            path: &str,
+            toolchain: &str,
+            extra_flags: &str,
+            all_targets: bool,
+            args: &[&str],
+        ) -> Result<()> {
+            let all_targets = all_targets.then_some("--all-targets");
+            cmd(
+                "cargo",
+                chain!(
+                    [&toolchain, "check", extra_flags, "--manifest-path", path,],
+                    all_targets,
+                    args.into_iter().cloned()
+                ),
+            )
+            .run()?;
+            Ok(())
+        }
+        // Check, and let caller control flags.
+        let toolchain = self.get_toolchain_override();
+        let miri_manifest = path!(self.miri_dir / "Cargo.toml");
+        let cargo_miri_manifest = path!(self.miri_dir / "cargo-miri" / "Cargo.toml");
+        let args = flags.iter().map(String::as_str).collect::<Vec<_>>();
+        check_package(
+            &miri_manifest.to_string_lossy().to_owned(),
```

Comment:
```suggestion
            &miri_manifest.to_string_lossy().into_owned(),
```
also applies for other `Cow`s